core/lib/Thelia/Core/Template/Assets/AsseticAssetManager.php

```diff
@@ -212,39 +214,18 @@ protected function decodeAsseticFilters(FilterManager $filterManager, $filters)
             foreach ($filter_list as $filter_name) {
                 $filter_name = trim($filter_name);
 
-                switch ($filter_name) {
-                    case 'less':
-                        $filterManager->set('less', new LessDotPhpFilter());
-                        break;
-
-                    case 'less_legacy':
-                        $filterManager->set('less_legacy', new Filter\LessphpFilter());
-                        break;
-
-                    case 'sass':
-                        $filterManager->set('sass', new Filter\Sass\SassFilter());
-                        break;
-
-                    case 'cssembed':
-                        $filterManager->set('cssembed', new Filter\PhpCssEmbedFilter());
-                        break;
-
-                    case 'cssrewrite':
-                        $filterManager->set('cssrewrite', new Filter\CssRewriteFilter());
-                        break;
-
-                    case 'cssimport':
-                        $filterManager->set('cssimport', new Filter\CssImportFilter());
-                        break;
+                foreach ($this->assetFilters as $filterIdentifier => $filterInstance) {
+                    if ($filterIdentifier == $filter_name) {
+                        $filterManager->set($filterIdentifier, $filterInstance);
+                    }
+                    // No, goto is not evil.
+                    goto filterFound;
```

Comment:
I don't know if it's evil or not but are you sure to use it here ? Why don't in the if statement just before ?

Here I will always go to filterfound, don't you think ?